packages/editor/src/components/editor/FormToolbar.tsx

```diff
@@ -40,19 +44,24 @@ export const FormToolbar = () => {
       }
       return { ...old, responsiveMode: next };
     });
+
+  const [elementsPanelOpen, setElementsPanelOpen] = useState(false);
+  const [structurePanelOpen, setStructurePanelOpen] = useState(false);
+  const [actionPanelOpen, setActionPanelOpen] = useState(false);
+  const { active } = useDndContext();
+
+  useEffect(() => {
+    if (active !== undefined) {
+      setElementsPanelOpen(false);
+      setStructurePanelOpen(false);
+      setActionPanelOpen(false);
+    }
+  }, [active]);
```

Comment:
Can't you move this open states and the check for an active dnd element into the PaletteCategoryPopover itself?

Reply:
yes, you're right, thanks 👍